swap/swap.go

```diff
@@ -542,7 +546,18 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-// Deploy deploys the Swap contract
+// NewInstanceAt creates a new instance of the chequebook contract at address and sets chequebookAddr
+func (s *Swap) NewInstanceAt(address common.Address, backend swap.Backend) error {
+	c, err := contract.InstanceAt(address, backend)
```

Comment:
why not right away
`s.contract, err`
No need for an intermediate variable

Reply:
Done